### ISSUE TITLE
InputEventKey.scancode does not return the correct value on non-QWERTY layouts

### ISSUE THREAD

**Issue by JapanYoshi** [godotengine/godot#46751]
**Godot version:** 3.2.3.stable.mono.official

**OS/device including version:**
Windows 10

**Issue description:**
I use the Dvorak keyboard layout, which has a different key placement from QWERTY.

For example, in JavaScript, there are different values in a keyboard event that correspond to the *physical* key pressed (the "scancode" in its original sense), and to the *character* corresponding to the key pressed.

However, on Godot, `InputEventKey.scancode` does not return the scancode on Dvorak, but the scancode of the key that corresponds to the character on QWERTY.

Here is the result of pressing the Q key on QWERTY:
![image](https://user-images.githubusercontent.com/8707990/110236385-e2095e80-7f78-11eb-93ae-51eb889dd216.png)

And here is the result of pressing the same key on Dvorak:
![image](https://user-images.githubusercontent.com/8707990/110236393-f188a780-7f78-11eb-942f-41f9e5d1ff76.png)
It should say this instead:
```
scancode: 0d0081 = 0x0051 (Q)
unicode: 0d0039 = 0x0027 (')
get_scancode_with_modifiers():
 81 (0x00000051)
```

For reference, here's the result of pressing the same key on ЙЦУКЕН, the standard keyboard layout for Russian:
![image](https://user-images.githubusercontent.com/8707990/110236442-36144300-7f79-11eb-9155-2921d0a24695.png)
(As a side note, this reportedly (#46750) does not work on Linux.)

**Steps to reproduce:**

1. Listen for any `InputEventKey`.
2. Inspect its properties.

**Minimal reproduction project:**
[Test.zip](https://github.com/godotengine/godot/files/6096913/Test.zip)

**Comment by Calinou** [comment]
As I said on the Godot Forums thread, I'm not sure if this can be fixed before 4.0 is released for compatibility reasons.

**Comment by bruvzg** [comment]
>  I'm not sure if this can be fixed before 4.0 is released for compatibility reasons.

With exception to `scancode` → `keycode` rename, there's no breaking changes, it was originally developed for 3.2.

Here's backport without renames: #46764

**Comment by madmiraal** [comment]
Fixed on master with #18020 and on 3.x with #46764.
